### PR TITLE
[IOTDB-5284] Fix some code smells

### PR DESCRIPTION
**Pull request by BUAAserein**
### Fix some code smells in server [iotdb-server], please see in [JIRA5284](https://issues.apache.org/jira/browse/IOTDB-5284)

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/query/control/FileReaderManager.java [hunk 1]
```diff
@@ -246,7 +246,7 @@ public synchronized boolean contains(TsFileResource tsFile, boolean isClosed) {
   public synchronized void writeFileReferenceInfo() {
     DEBUG_LOGGER.info("[closedReferenceMap]\n");
     for (Map.Entry<String, AtomicInteger> entry : closedReferenceMap.entrySet()) {
-      DEBUG_LOGGER.info(String.format("\t%s: %d\n", entry.getKey(), entry.getValue().get()));
+      DEBUG_LOGGER.info(String.format("\t%s: %d%n", entry.getKey(), entry.getValue().get()));
```
Comment (SpriCoder) [inline server/src/main/java/org/apache/iotdb/db/query/control/FileReaderManager.java:249]: Why change `\n` to `%n` here?

**Review by SpriCoder — CHANGES_REQUESTED** [review]
PTAL~

File: server/src/main/java/org/apache/iotdb/db/query/control/FileReaderManager.java (hunk 1 above)
Comment (BUAAserein) [reply server/src/main/java/org/apache/iotdb/db/query/control/FileReaderManager.java:249]: Please check https://rules.sonarsource.com/java/RSPEC-3457 and https://stackoverflow.com/questions/31981136/difference-between-n-and-n-for-printing-a-new-line-in-java

Comment (BUAAserein) [reply server/src/main/java/org/apache/iotdb/db/query/control/FileReaderManager.java:249]: I just learned that %n should be used in place of \n to produce the platform-specific line separator

Comment (SpriCoder) [reply server/src/main/java/org/apache/iotdb/db/query/control/FileReaderManager.java:249]: Get it~

**Review by SpriCoder — APPROVED** [review]
LGTM~

File: server/src/main/java/org/apache/iotdb/db/service/SettleService.java [hunk 2]
```diff
@@ -123,9 +123,9 @@ public void startSettling(
       return;
     }
     logger.info(
-        "Totally find "
-            + (seqResourcesToBeSettled.size() + unseqResourcesToBeSettled.size())
-            + " tsFiles to be settled.");
+        "Totally find {}{} tsFiles to be settled.",
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/service/SettleService.java:126]: should plus ?

Comment (BUAAserein) [reply server/src/main/java/org/apache/iotdb/db/service/SettleService.java]: Already fixed

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**